runtime/src/main/java/org/corfudb/runtime/object/transactions/AbstractTransactionalContext.java

```diff
@@ -276,23 +276,14 @@ void addToWriteSet(ICorfuSMRProxy proxy, SMREntry updateEntry, Object[] conflict
 
     void mergeWriteSetInto(Map<UUID, AbstractMap.SimpleEntry<Set<Integer>, List<SMREntry>>> otherWSet) {
         otherWSet.entrySet().forEach(e-> {
-            // create an entry for this streamID
-            writeSet.computeIfAbsent(e.getKey(),                    // the streamID
-                    u -> new AbstractMap.SimpleEntry<>(new HashSet<>(), new ArrayList<>() ));
-
-            // copy all the conflict-params set for this streamID
-            writeSet.get(e.getKey())                 // the entry pair for this streamID
-                    .getKey()                        // the left componentof the pair is a conflict-param set
-                    .addAll(e.getValue()             // the pair from the otherWSet
-                            .getKey());              // the left component of the pair
-
-            // copy all the WriteSetEntry list for this streamID
-            writeSet.get(e.getKey())                 // the entry pair for this streamID
-                    .getValue()                      // the right componentof the pair is a WriteSetEntry list
-                    .addAll(e.getValue()             // the pair from the otherWSet
-                            .getValue());            // the right component of the pair
-
 
+            // Since nested transactions inherit the write set of its parent, in order to merge the
+            // write set of the child transaction, we only need to add new entries for non-existing
+            // streams.
+            // For entries that were inherited, they share the same reference in the write set map. Any
+            // modification of it will also appear in the parent write set.
+            writeSet.computeIfAbsent(e.getKey(),
```

Comment:
Hm, I wonder if this is a correctness issue? What if the nested transaction aborts? Does it's write set get removed?